subprojects/docs/src/docs/userguide/troubleshooting/validation_problems.adoc

```diff
@@ -210,5 +210,57 @@ If you use the `@Input` annotation on a file-based property, Gradle wouldn't con
 To fix this problem, you need to tell Gradle if the file property represents an input file, in which case you should annotate it with `@InputFile`, or a directory, in which case it should be annotated with `@InputDirectory`.
 If what you really wanted to say is that the actual file path is an input, then you should return a `String` instead which corresponds to the absolute path of the file.
 
+[[unresolvable_input]]
+== Unresolvable input
 
+This error indicates that a file collection is used as an input of a task, but that resolution failed, for example because of the use of an invalid notation for a file.
+Previous versions of Gradle used to skip validation if a task didn't declare any output but task execution would have failed.
+In practice, it means that would have been a failure at execution time is a failure when inputs are validated.
```

Comment:
```suggestion
In practice, it means that resolving the input file collection would have been a failure at execution time is a failure when inputs are validated.
```

It doesn't capture completely what is happening there. This is what worked in Gradle 6.8:
- The task declares some input via a file collection
- Gradle determines the (task) dependencies of that file collection and that works
- The task action doesn't resolve the input file collection, so the task passes if it doesn't declare any outputs (so we don't snapshot the input files).

With Gradle 7.0, we started resolving the roots of the input file collections even for tasks with no outputs. And that breaks tasks which did the above. So we capture the failure and emit a deprecation warning. And in Gradle 8.0 all input file collection need to be resolvable when you run the task, no matter if the task declares outputs or not.